src/App.js

```diff
@@ -1,22 +1,58 @@
-import { useState } from "react";
-import { teamDetails } from "../src/assests/teamDetails";
-import CardComponent from "./components/Card";
+import { useState, useEffect } from "react";
+
 import Header from "./components/Header";
+import userData from "./assets/userData.json";
 import "./App.css";
-
-const CardContainer = ({ searchedTeamDetails }) => {
-  return searchedTeamDetails.map((member) => {
-    return <CardComponent member={member} key={member.id} />;
-  });
-};
+import NoResultsComponent from "./components/NoResults/NoResultsComponent";
+import { CardContainer } from "./components/CardContainer";
 
 const App = () => {
-  const [searchedTeamDetails, setSearchedTeamDetails] = useState(teamDetails);
+  const [teamDetails, setTeamDetails] = useState([]);
+  const [searchedTeamDetails, setSearchedTeamDetails] = useState([]);
+  const [noResult, setNoResult] = useState(false);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  function fetchData() {
+    let teamArr = [];
```

Comment:
```suggestion
    const teamArr = [];
```
Since you are just pushing new elements to this array and it's not getting reassigned, you can declare it using const.